### PR TITLE
Validate owning application, if this functionality is provided by plugins.

### PR DESCRIPTION
**Pull request by antban**
Right now event type evolution includes modification of different fields. And nakadi may have some plugins, that are allowing this evolution to proceed. 

Unfortunately, the plugins where not actually used on event-type/subscription creation/update. This PR reintroduces owning_application validation for event types and subscriptions. 

In order to have flawless roll out, the both checks are guarded with feature toggles.

Also, some tests were cleaned up

### CODE REVIEW

**Comment by antban** [comment]
:+1:

File: api-metastore/src/main/java/org/zalando/nakadi/controller/EventTypeController.java [hunk 1]
```diff
@@ -90,11 +90,10 @@ public void initBinder(final WebDataBinder binder) {
 
     @RequestMapping(method = RequestMethod.POST)
     public ResponseEntity<?> create(@Valid @RequestBody final EventTypeBase eventType,
-                                    final Errors errors,
-                                    final NativeWebRequest request)
+                                    final Errors errors)
             throws TopicCreationException, InternalNakadiException, NoSuchPartitionStrategyException,
             DuplicatedEventTypeNameException, InvalidEventTypeException, ValidationException,
-            FeatureNotAvailableException {
+            FeatureNotAvailableException, WrongOwningApplicationException {
```
Comment (a1exsh) [inline api-metastore/src/main/java/org/zalando/nakadi/controller/EventTypeController.java:96]: Can we name it `InvalidOwningApplicationException` instead?  I believe "wrong" can be interpreted as valid, but not the expected one, though we can't have an expectation here: we don't know better than the user.

File: api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionService.java [hunk 2]
```diff
@@ -166,7 +168,7 @@ public Subscription updateSubscription(final String subscriptionId, final Subscr
 
         authorizationValidator.authorizeSubscriptionAdmin(old);
 
-        subscriptionValidationService.validateSubscriptionChange(old, newValue);
+        subscriptionValidationService.validateSubscriptionOnChange(old, newValue);
```
Comment (a1exsh) [inline api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionService.java:171]:
```suggestion
        subscriptionValidationService.validateSubscriptionOnUpdate(old, newValue);
```

If it's called `CRUD`...

File: api-metastore/src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java [hunk 3]
```diff
@@ -311,7 +327,55 @@ public void whenEventTypeUpdatedThenKPIEventSubmitted() throws Exception {
     }
 
     @Test
-    public void whenEventTypeDeletedThenKPIEventSubmitted() throws Exception {
+    public void whenEventTypeOwningApplicationUpdatedThanItIsValidated() {
```
Comment (a1exsh) [inline api-metastore/src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java:330]:
```suggestion
    public void whenEventTypeOwningApplicationUpdatedThenItIsValidated() {
```

Comment (a1exsh) [reply api-metastore/src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java:330]: See also the few next test names.

**Comment by a1exsh** [comment]
`<=` https://github.com/zalando/nakadi/pull/1309

**Comment by a1exsh** [comment]
:+1:

**Comment by thomasabraham** [comment]
I'm resolving the conflicts.

**Comment by a1exsh** [comment]
:+1:

**Comment by thomasabraham** [comment]
:+1:

**Comment by antban** [comment]
:-1:

**Comment by antban** [comment]
:+1:

File: api-consumption/src/main/java/org/zalando/nakadi/controller/advice/SubscriptionStreamExceptionHandler.java [hunk 4]
```diff
@@ -19,9 +19,10 @@
 @ControllerAdvice(assignableTypes = SubscriptionStreamController.class)
 public class SubscriptionStreamExceptionHandler implements AdviceTrait {
 
-    @ExceptionHandler(WrongStreamParametersException.class)
-    public ResponseEntity<Problem> handleWrongStreamParametersException(final WrongStreamParametersException exception,
-                                                                        final NativeWebRequest request) {
+    @ExceptionHandler(InvalidStreamParametersException.class)
+    public ResponseEntity<Problem> handleWrongStreamParametersException(
```
Comment (a1exsh) [inline api-consumption/src/main/java/org/zalando/nakadi/controller/advice/SubscriptionStreamExceptionHandler.java:23]:
```suggestion
    public ResponseEntity<Problem> handleInvalidStreamParametersException(
```

**Comment by a1exsh** [comment]
:+1:

**Comment by antban** [comment]
:+1:

**Pull request merged**